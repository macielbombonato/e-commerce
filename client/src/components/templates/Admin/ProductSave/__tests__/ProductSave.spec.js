import createTestContext from 'react-cosmos-test/enzyme';

import fixture from '../ProductSave.fixture';

const { mount, getWrapper, get } = createTestContext({ fixture });

beforeEach(mount);

describe('#ProductSave template', () => {
  it('renders component', () => {
    expect(getWrapper()).toMatchSnapshot();
  });

  it('cancels the product creation and gets the user back to product listing page', () => {
    getWrapper('.btn-secondary').simulate('click');

    expect(get('props').history.push).toBeCalledWith('/admin/product');
  });

  it('saves the product creation and gets the user back to product listing page', async () => {
    const inputName = getWrapper('input#name').first();
    const inputImage = getWrapper('input#image').first();

    inputName.simulate('change', { target: { value: 'product 1' } });
    inputImage.simulate('change', { target: { files: [] } });
    inputImage.simulate('change', { target: { files: ['dummyValue.something'] } });

    await getWrapper('.btn-success').simulate('click');

    expect(get('props').saveProduct).toBeCalled();
    expect(get('props').history.push).toBeCalledWith('/admin/product');
  });
});
