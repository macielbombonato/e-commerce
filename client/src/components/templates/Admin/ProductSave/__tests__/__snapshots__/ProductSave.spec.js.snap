// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ProductSave template renders component 1`] = `
<ProductSave
  history={
    Object {
      "push": [Function],
    }
  }
  saveProduct={[Function]}
>
  <Container
    tag="div"
  >
    <div
      className="container"
    >
      <h1
        className="display-4"
      >
        Product Save
      </h1>
      <Form
        tag="form"
      >
        <form
          className=""
        >
          <FormGroup
            tag="div"
          >
            <div
              className="form-group"
            >
              <Label
                for="name"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <label
                  className=" form-control-label"
                  htmlFor="name"
                >
                  Name
                </label>
              </Label>
              <Input
                id="name"
                name="name"
                onChange={[Function]}
                tag="p"
                type="text"
              >
                <input
                  className="form-control"
                  id="name"
                  name="name"
                  onChange={[Function]}
                  type="text"
                />
              </Input>
            </div>
          </FormGroup>
          <FormGroup
            tag="div"
          >
            <div
              className="form-group"
            >
              <Label
                for="description"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <label
                  className=" form-control-label"
                  htmlFor="description"
                >
                  Description
                </label>
              </Label>
              <Input
                id="description"
                name="description"
                onChange={[Function]}
                tag="p"
                type="textarea"
              >
                <textarea
                  className="form-control"
                  id="description"
                  name="description"
                  onChange={[Function]}
                />
              </Input>
            </div>
          </FormGroup>
          <FormGroup
            tag="div"
          >
            <div
              className="form-group"
            >
              <Label
                for="price"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <label
                  className=" form-control-label"
                  htmlFor="price"
                >
                  Price
                </label>
              </Label>
              <Input
                id="price"
                name="price"
                onChange={[Function]}
                tag="p"
                type="number"
              >
                <input
                  className="form-control"
                  id="price"
                  name="price"
                  onChange={[Function]}
                  type="number"
                />
              </Input>
            </div>
          </FormGroup>
          <FormGroup
            tag="div"
          >
            <div
              className="form-group"
            >
              <Label
                for="factor"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <label
                  className=" form-control-label"
                  htmlFor="factor"
                >
                  Factor
                </label>
              </Label>
              <Input
                id="factor"
                name="factor"
                onChange={[Function]}
                tag="p"
                type="select"
              >
                <select
                  className="form-control"
                  id="factor"
                  name="factor"
                  onChange={[Function]}
                >
                  <option
                    value="A"
                  >
                    A
                  </option>
                  <option
                    value="B"
                  >
                    B
                  </option>
                  <option
                    value="C"
                  >
                    C
                  </option>
                </select>
              </Input>
            </div>
          </FormGroup>
          <FormGroup
            tag="div"
          >
            <div
              className="form-group"
            >
              <Label
                for="image"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <label
                  className=" form-control-label"
                  htmlFor="image"
                >
                  Image
                </label>
              </Label>
              <Input
                id="image"
                name="image"
                onChange={[Function]}
                tag="p"
                type="file"
              >
                <input
                  className="form-control-file"
                  id="image"
                  name="image"
                  onChange={[Function]}
                  type="file"
                />
              </Input>
            </div>
          </FormGroup>
          <div
            className="clearfix"
          >
            <Button
              className="float-left"
              color="secondary"
              onClick={[Function]}
              tag="button"
            >
              <button
                className="float-left btn btn-secondary"
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
            </Button>
            <Button
              className="float-right"
              color="success"
              onClick={[Function]}
              tag="button"
            >
              <button
                className="float-right btn btn-success"
                onClick={[Function]}
                type="button"
              >
                Submit
              </button>
            </Button>
          </div>
        </form>
      </Form>
    </div>
  </Container>
</ProductSave>
`;
